src/main/java/org/terracotta/statistics/TableValueStatistic.java

```diff
@@ -0,0 +1,102 @@
+/*
+ * Copyright Terracotta, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.terracotta.statistics;
+
+import org.terracotta.statistics.extended.StatisticType;
+
+import java.io.Serializable;
+import java.util.Collection;
+import java.util.function.BiConsumer;
+import java.util.function.Consumer;
+
+/**
+ * A statistic that can dynamically construct and provide a table statistic based on provided suppliers
+ *
+ * @author Mathieu Carbou
+ */
+public class TableValueStatistic extends TableSkeleton implements ValueStatistic<Table> {
```

Comment:
Why is there inheritance here?  Is intentional or just convenience - composition would seem more natural to me.

Reply:
intentional: `TableValueStatistic` and `Table` are both tables of ValueStatistic, the only difference is that one is fixed in time and represent a table of constant stats, and the other one is a stat providing a Table, and it contains several ValueStatistic, that could delegate the value retrieval to a supplier. There is a test that shows the difference.